tracee-ebpf/tracee/net_events.go

```diff
@@ -16,162 +15,30 @@ func (t *Tracee) processNetEvents() {
 	for {
 		select {
 		case in := <-t.netChannel:
-			// Sanity check - timestamp, event id, host tid and comm must exist in all net events
+			// Sanity check - timestamp, event id, host tid and processName must exist in all net events
 			if len(in) < 32 {
 				continue
 			}
+			evtMeta, dataBuff := parseEventMetaData(in)
 
-			timeStamp := binary.LittleEndian.Uint64(in[0:8])
-			netEventId := binary.LittleEndian.Uint32(in[8:12])
-			hostTid := binary.LittleEndian.Uint32(in[12:16])
-			comm := string(bytes.TrimRight(in[16:32], "\x00"))
-			dataBuff := bytes.NewBuffer(in[32:])
-
-			// timeStamp is nanoseconds since system boot time
-			timeStampObj := time.Unix(0, int64(timeStamp+t.bootTime))
+			processContext, exist := t.getProcessCtx(evtMeta.HostTid)
+			if exist != nil {
+				t.handleError(fmt.Errorf("couldn't find the process: %d", evtMeta.HostTid))
+				continue
+			}
 
-			if netEventId == NetPacket {
-				var pktLen uint32
-				err := binary.Read(dataBuff, binary.LittleEndian, &pktLen)
-				if err != nil {
-					t.handleError(err)
-					continue
-				}
-				var ifindex uint32
-				err = binary.Read(dataBuff, binary.LittleEndian, &ifindex)
-				if err != nil {
-					t.handleError(err)
-					continue
-				}
-				idx, ok := t.ngIfacesIndex[int(ifindex)]
-				if !ok {
-					t.handleError(err)
-					continue
-				}
+			eventName := EventsDefinitions[evtMeta.NetEventId].Name
+			evt, ShouldCapture, cap := network_protocols.ProcessNetEvent(dataBuff, evtMeta, eventName, processContext)
+			t.config.ChanEvents <- evt
+			t.stats.eventCounter.Increment()
 
-				if t.config.Debug {
-					var pktMeta struct {
-						SrcIP    [16]byte
-						DestIP   [16]byte
-						SrcPort  uint16
-						DestPort uint16
-						Protocol uint8
-						_        [3]byte //padding
-					}
-					err = binary.Read(dataBuff, binary.LittleEndian, &pktMeta)
-					if err != nil {
+			if ShouldCapture {
+				interfaceIndex, ok := t.ngIfacesIndex[int(cap.InterfaceIndex)]
+				if ok {
+					if err := t.writePacket(cap.PacketLen, time.Unix(int64(evt.Timestamp), 0), interfaceIndex, dataBuff); err != nil {
```

Comment:
can you explain why changing the time object? it used to be created like so: `time.Unix(0, int64(timeStamp+t.bootTime))`

Reply:
Because of  the time that came from the event is more accurate then the time in the user-space

Reply:
yes but the time that comes from the event is in nanoseconds since system boot time (it comes from `bpf_ktime_get_ns()`). or am i missing something here?